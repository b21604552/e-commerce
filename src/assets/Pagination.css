.pagination-wrapper {
    height: 33px;
    display: inline-flex;
    margin-bottom: 12px;
}

.pagination-element {
    width: 33px;
    height: 33px;
    background: #FFFFFF;
    border: 1px solid #EEEEEE;
    box-sizing: border-box;
    border-radius: 9px;
    font-style: normal;
    font-weight: normal;
    font-size: 10px;
    line-height: 20px;
    display: flex;
    align-items: center;
    justify-content: center;
    color: #646464;
    margin-right: 6px;
    cursor: pointer;
}

.pagination-element-active {
    color: #FF6000;
}

.pagination-prev {
    margin-left: 13px;
}

.pagination-next {
    margin-right: 13px;
}