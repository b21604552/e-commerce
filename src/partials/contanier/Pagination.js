import React, { useEffect, useState } from 'react'

export default function Pagination (props) {
  const [pagination, setPagination] = useState(props.pagination)

  useEffect(() => {
    setPagination(props.pagination)
  }, [props.pagination])

  return (
    <div className={'pagination-wrapper'}>
      {
        pagination !== null && (
          <>

            {pagination.isPrevEnable && (
              <div id={'prev-page'} className={'pagination-element pagination-prev'} onClick={() => props.setPage(pagination.currentPage - 1)}>&lt;</div>
            )}

            {pagination.pages.length > 0 && (
              <>
                {
                  pagination.pages.map((page) => (
                    <div id={'page-' + page.key} key={'page-' + page.key} className={page.active ? ('pagination-element pagination-element-active') : ('pagination-element')}
                         onClick={() => props.setPage(page.key)}>{page.key}</div>
                  ))
                }
              </>
            )}

            {pagination.isNextEnable && (
              <div id={'next-page'} className={'pagination-element pagination-next'} onClick={() => props.setPage(pagination.currentPage + 1)}>></div>
            )}

          </>
        )
      }
    </div>
  )
}