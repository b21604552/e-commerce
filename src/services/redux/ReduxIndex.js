import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
