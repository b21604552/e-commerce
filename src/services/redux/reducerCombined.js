import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
