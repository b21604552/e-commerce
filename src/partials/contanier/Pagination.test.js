import React from 'react'
import ReactDOM from 'react-dom'
import Pagination from './Pagination'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({ adapter: new Adapter() })

describe('Pagination Test', () => {

  it('Renders Pagination without crashing', () => {
    const div = document.createElement('div')

    const pagination = {
      isPrevEnable: false,
      isNextEnable: true,
      currentPage: 1,
      pages: [{ active: true, key: 1 }, { active: false, key: 2 }, { active: false, key: 3 }, { active: false, key: 4 }]
    }

    ReactDOM.render(
      <Provider store={reduxStore}>
        <Pagination pagination={pagination}/>
      </Provider>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders Pagination correctly', () => {
    const pagination = {
      isPrevEnable: false,
      isNextEnable: true,
      currentPage: 1,
      pages: [{ active: true, key: 1 }, { active: false, key: 2 }, { active: false, key: 3 }, { active: false, key: 4 }]
    }

    const div = renderer.create(
      <Provider store={reduxStore}>
        <Pagination pagination={pagination}/>
      </Provider>).toJSON()
    expect(div).toMatchSnapshot()
  })

  it('Next button test', () => {
    const pagination = {
      isPrevEnable: false,
      isNextEnable: true,
      currentPage: 1,
      pages: [{ active: true, key: 1 }, { active: false, key: 2 }, { active: false, key: 3 }, { active: false, key: 4 }]
    }

    function setPage (page) {
      expect(page).toEqual(2)
    }

    const comp = shallow(<Pagination pagination={pagination} setPage={(page) => setPage(page)}/>)
    comp.find('#next-page').simulate('click')
  })

  it('Prev button test', () => {
    const pagination = {
      isPrevEnable: true,
      isNextEnable: false,
      currentPage: 2,
      pages: [{ active: false, key: 1 }, { active: true, key: 2 }, { active: false, key: 3 }, { active: false, key: 4 }]
    }

    function setPage (page) {
      expect(page).toEqual(1)
    }

    const comp = shallow(<Pagination pagination={pagination} setPage={(page) => setPage(page)}/>)
    comp.find('#prev-page').simulate('click')
  })

  it('Page button test', () => {
    const pagination = {
      isPrevEnable: true,
      isNextEnable: false,
      currentPage: 2,
      pages: [{ active: false, key: 1 }, { active: true, key: 2 }, { active: false, key: 3 }, { active: false, key: 4 }]
    }

    function setPage (page) {
      expect(page).toEqual(3)
    }

    const comp = shallow(<Pagination pagination={pagination} setPage={(page) => setPage(page)}/>)
    comp.find('#page-3').simulate('click')
  })

})