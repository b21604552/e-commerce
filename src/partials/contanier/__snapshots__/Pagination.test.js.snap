// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Test Renders Pagination correctly 1`] = `
<div
  className="pagination-wrapper"
>
  <div
    className="pagination-element pagination-element-active"
    id="page-1"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="pagination-element"
    id="page-2"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="pagination-element"
    id="page-3"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="pagination-element"
    id="page-4"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="pagination-element pagination-next"
    id="next-page"
    onClick={[Function]}
  >
    &gt;
  </div>
</div>
`;
